### PR TITLE
Apply PR 2289: Update basic method signature in reference doc

### PR DESCRIPTION
**Pull request by pshao25**
Apply https://github.com/Azure/autorest.csharp/pull/2289

### CODE REVIEW

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/src/Generated/DeviceUpdateClient.cs [hunk 1]
```diff
@@ -1274,9 +1310,11 @@ IEnumerable<Page<BinaryData>> CreateEnumerable(string nextLink, int? pageSizeHin
...
         /// <param name="action"> Import update action. Allowed values: &quot;import&quot;. </param>
-        /// <param name="content"> The content to send as the body of the request. </param>
-        /// <param name="context"> The request context, which can override default behaviors on the request on a per-call basis. </param>
+        /// <param name="content"> The content to send as the body of the request. Details of the request body schema are in the Remarks section below. </param>
+        /// <param name="context"> The request context, which can override default behaviors of the client pipeline on a per-call basis. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="action"/> or <paramref name="content"/> is null. </exception>
+        /// <exception cref="RequestFailedException"> Service returned a non-success status code. </exception>
+        /// <returns cref="Task{Operation}"> The response returned from the service. Details of the response body schema are in the Remarks section below. </returns>
```
Comment (annelo-msft) [inline sdk/deviceupdate/Azure.IoT.DeviceUpdate/src/Generated/DeviceUpdateClient.cs:1317]: This method returns: `Task<Operation<BinaryData>>`

Comment (annelo-msft) [inline sdk/deviceupdate/Azure.IoT.DeviceUpdate/src/Generated/DeviceUpdateClient.cs:1317]: The text should reflect the return value of the method. I recommend:

`The <see cref="Operation{T}"/> from the service that will contain a <see cref="BinaryData"/> object once the asynchronous operation on the service has completed. Details of the body schema for the operation's final value are in the Remarks section below.`

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/src/Generated/DeviceUpdateClient.cs [hunk 2]
```diff
@@ -1464,9 +1504,11 @@ public virtual Operation<BinaryData> ImportUpdate(WaitUntil waitUntil, string ac
         /// <param name="provider"> Update provider. </param>
         /// <param name="name"> Update name. </param>
         /// <param name="version"> Update version. </param>
-        /// <param name="context"> The request context, which can override default behaviors on the request on a per-call basis. </param>
+        /// <param name="context"> The request context, which can override default behaviors of the client pipeline on a per-call basis. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="provider"/>, <paramref name="name"/> or <paramref name="version"/> is null. </exception>
         /// <exception cref="ArgumentException"> <paramref name="provider"/>, <paramref name="name"/> or <paramref name="version"/> is an empty string, and was expected to be non-empty. </exception>
+        /// <exception cref="RequestFailedException"> Service returned a non-success status code. </exception>
+        /// <returns cref="Task{Operation}"> The response returned from the service. </returns>
```
Comment (annelo-msft) [inline sdk/deviceupdate/Azure.IoT.DeviceUpdate/src/Generated/DeviceUpdateClient.cs:1511]: The text should reflect the return value of the method. I recommend:

`The <see cref="Operation"/> representing an asynchronous operation on the service.`

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/src/Generated/DeviceUpdateClient.cs (hunk 1 above)
Comment (pshao25) [reply sdk/deviceupdate/Azure.IoT.DeviceUpdate/src/Generated/DeviceUpdateClient.cs:1317]: We cannot generate this in comments with two-level generic type. It is a known issue, I saw many discussion in github:
https://github.com/dotnet/csharplang/issues/401
https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/2304

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/src/Generated/DeviceUpdateClient.cs (hunk 2 above)
Comment (pshao25) [reply sdk/deviceupdate/Azure.IoT.DeviceUpdate/src/Generated/DeviceUpdateClient.cs:1511]: How about
`/// <returns>An <see cref="Operation"/> representing an asynchronous operation on the service.</returns>`

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/src/Generated/DeviceUpdateClient.cs (hunk 1 above)
Comment (pshao25) [reply sdk/deviceupdate/Azure.IoT.DeviceUpdate/src/Generated/DeviceUpdateClient.cs:1317]: How about 
```
/// <returns>An <see cref="Operation{T}"/> from the service that will contain a <see cref="BinaryData"/> object once the asynchronous operation on the service has completed. Details of the body schema for the operation's final value are in the Remarks section below.</returns>
```

Comment (pshao25) [reply sdk/deviceupdate/Azure.IoT.DeviceUpdate/src/Generated/DeviceUpdateClient.cs:1317]: According to [this](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments#d313-returns), how about we remove the `cref` part?

Comment (pshao25) [reply sdk/deviceupdate/Azure.IoT.DeviceUpdate/src/Generated/DeviceUpdateClient.cs:1317]: An explanation of [CDATA] is 

If we generate like this
/// <returns> The Pageable<BinaryData>
The build fails.

If we generate like this
/// <returns> The Pageable{BinaryData}
The rendered doc is like
image

Comment (pshao25) [reply sdk/deviceupdate/Azure.IoT.DeviceUpdate/src/Generated/DeviceUpdateClient.cs:1317]: Why we can't put `<see cref="Operation{T}"/>`  grammer in text?
We have two-level type [Operation<AsyncPageable<BinaryData>>](https://github.com/archerzz/autorest.csharp/blob/00d2093b0d2d594fc23fdab7d63f1fb8f31cf81a/test/TestServerProjectsLowLevel/paging/Generated/PagingClient.cs#L1712).

Comment (pshao25) [reply sdk/deviceupdate/Azure.IoT.DeviceUpdate/src/Generated/DeviceUpdateClient.cs]: See preview here:
https://raw.githack.com/Azure/azure-sdk-for-net/1a0c08bc10/doc/ApiDocGeneration/Azure.AI.Language.Conversations/docfx-output/docfx_project/site/api/index.html
https://raw.githack.com/Azure/azure-sdk-for-net/1a0c08bc10/doc/ApiDocGeneration/Azure.IoT.DeviceUpdate/docfx-output/docfx_project/site/api/index.html
https://github.com/Azure/azure-sdk-for-net/blob/1a0c08bc10/doc/ApiDocGeneration/Azure.Messaging.WebPubSub/docfx-output/docfx_project/site/api/index.html
https://github.com/Azure/azure-sdk-for-net/blob/1a0c08bc10/doc/ApiDocGeneration/Azure.Security.ConfidentialLedger/docfx-output/docfx_project/site/api/index.html

**Review by ArcturusZhang — APPROVED** [review]

**Comment by chunyu3** [comment]
better to remove the changes of Azure.AI.Language.QuestionAnswering since it is excluded from build.

**Pull request merged**